packages/vector_graphics/lib/src/render_vector_graphics.dart

```diff
@@ -135,18 +176,14 @@ class RenderVectorGraphic extends RenderBox {
   Future<void>? _pendingRasterUpdate;
   bool _disposed = false;
 
-  // Re-create the raster for a given vector graphic if the target size
-  // is sufficiently different.
-  Future<void> _maybeUpdateRaster(Size desiredSize) async {
-    final int scaledWidth =
-        (pictureInfo.size.width * devicePixelRatio / scale).round();
-    final int scaledHeight =
-        (pictureInfo.size.height * devicePixelRatio / scale).round();
-    if (_lastRasterizedSize != null &&
-        _lastRasterizedSize!.width == scaledWidth &&
-        _lastRasterizedSize!.height == scaledHeight) {
-      return;
+  static Future<_RasterData> _createRaster(_RasterKey key, double scaleFactor) {
+    if (_pendingRasterCache.containsKey(key)) {
+      return _pendingRasterCache[key]!;
     }
+
+    final PictureInfo info = key.info;
```

Comment:
This should be what the key looks up rather than part of the key.

Reply:
Alternatively, this shouldn't be part of the key at all since the cache is just holding images?